### ISSUE TITLE
brew list --cask shows symlinks of renamed packages

### ISSUE THREAD

**Issue by yochem** [Homebrew/brew#18849]
### `brew doctor` output

```shell
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 4.4.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e78a0adb4f260913fffdc956cc8c20fb54109feb
Last commit: 7 days ago
Branch: stable
Core tap JSON: 02 Dec 10:59 UTC
Core cask tap JSON: 02 Dec 10:59 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BUNDLE_USER_CACHE: /Users/yochem/Library/Caches/bundle
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.6 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 16.0.0 build 1600
Git: 2.45.0 => /usr/local/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.7.1-x86_64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
```

### What were you trying to do (and why)?

List all casks available on my system (to see if some could be removed)

### What happened (include all command output)?

```
$ brew list --cask -1
brave-browser
google-drive
hammerspoon
iterm2-beta
iterm2@beta
itsycal
logi-options+
logi-options-plus
microsoft-auto-update
microsoft-teams
monitorcontrol
spotify
telegram
utm
visual-studio-code
vlc
whatsapp
```

I thought, oh, I have both logi-options-plus and logo-options+ installed (same with iterm2). Better remove the former:

```
$ brew uninstall --cask logi-options-plus
Warning: Cask logi-options-plus was renamed to logi-options+.
```

Oh. I don't want to remove that one.

Let's check:

```
$ ls -l /usr/local/Caskroom
drwxr-xr-x 4 yochem admin 128 2024-07-20  brave-browser/
drwxr-xr-x 4 yochem admin 128 2024-05-09  google-drive/
drwxr-xr-x 4 yochem admin 128 2024-11-29  hammerspoon/
lrwxr-xr-x 1 yochem admin  11 2024-07-20  iterm2-beta -> 'iterm2@beta'/
drwxr-xr-x 4 yochem admin 128 2024-07-20 'iterm2@beta'/
drwxr-xr-x 4 yochem admin 128 2024-05-09  itsycal/
drwxr-xr-x 4 yochem admin 128 2024-10-25  logi-options+/
lrwxr-xr-x 1 yochem admin  13 2024-09-30  logi-options-plus -> logi-options+/
drwxr-xr-x 4 yochem admin 128 2024-05-09  microsoft-auto-update/
drwxr-xr-x 4 yochem admin 128 2024-05-09  microsoft-teams/
drwxr-xr-x 4 yochem admin 128 2023-09-17  monitorcontrol/
drwxr-xr-x 4 yochem admin 128 2024-05-09  spotify/
drwxr-xr-x 4 yochem admin 128 2024-05-09  telegram/
drwxr-xr-x 4 yochem admin 128 2024-05-09  utm/
drwxr-xr-x 4 yochem admin 128 2024-05-09  visual-studio-code/
drwxr-xr-x 4 yochem admin 128 2024-02-18  vlc/
drwxr-xr-x 4 yochem admin 128 2024-05-09  whatsapp/
```

Ah, symlinks, and the brew list command probably just lists the directory.

### What did you expect to happen?

`brew list --cask` shows exactly which casks I have installed and ignores links to the same cask.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
I'm unsure if you can still install casks with their old name, so to fake this situation:


$ brew install --cask logi-options+
$ ln -s $(brew --prefix)/Caskroom/logi-options+ $(brew --prefix)/Caskroom/sym-test
$ brew list --cask -1
logi-options+
sym-test
```
```

**Comment by yochem** [comment]
Probably need to add a symlink check [here](https://github.com/Homebrew/brew/blob/13c3def0496cc9af74a754187994361a8be86f0f/Library/Homebrew/cmd/list.rb#L217).

**Comment by MikeMcQuaid** [comment]
> Probably need to add a symlink check [here](https://github.com/Homebrew/brew/blob/13c3def0496cc9af74a754187994361a8be86f0f/Library/Homebrew/cmd/list.rb#L217).

@yochem Could you try and open a pull request? [This document](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) should help and we're happy to walk you through anything else.

Thanks!

**Comment by vidishanevatia** [comment]
The documentation was super helpful, I tested it with your reproduction instruction and it works just fine. Opened a PR above. @yochem

**Comment by stevenlele** [comment]
This issue is not fixed because the changed codepath in #18946 only takes effect when `--full-name`, `--versions` or `--pinned` are passed.

https://github.com/Homebrew/brew/blob/5d630cc0cfa62806697ee90feba3bc72551b0ef8/Library/Homebrew/cmd/list.rb#L110-L112

Most of the time `ls` is called (which is [documented](https://docs.brew.sh/Manpage#list-ls-options-installed_formulainstalled_cask-) in the `--full-name` option):

https://github.com/Homebrew/brew/blob/5d630cc0cfa62806697ee90feba3bc72551b0ef8/Library/Homebrew/cmd/list.rb#L165-L168

###### Note to self: symlinks to uninstalled casks can be cleaned by running `brew cleanup`.

**Comment by MikeMcQuaid** [comment]
@stevenlele can you open a new issue please with all requested information given this was closed over a year ago? Thanks!

**Comment by texastoland** [comment]
I was debugging buresdv/Cork#613 and suspect this is relevant to any formula? In that case I imagine the correct fix would be to revert #18946, create a method to check if it's renamed (similar to `installed_as_dependency?`), and add logic to print it differently or filter it completely. Naively I imagine this either affects `bundle dump` or is already solved a different way there.

**Comment by MikeMcQuaid** [comment]
@texastoland as above: open a new issue.
